tests/utils/state-locks-filter

```diff
@@ -0,0 +1,398 @@
+                return str(group)
+
+        return ""
+
+    # Retrieve the times for each trace in the file
+    # For each trace, there is a list with the times for each test where the trace appears
+    def get_traces_times(self) -> dict[LockTrace, list[GroupTimes]]:
+        traces = dict[LockTrace, list[GroupTimes]]()
+        for group in self.groups:
+            group_traces = group.get_traces()
+            for trace in group_traces:
+                group_time = GroupTimes(
+                    group.get_name(),
+                    group.get_lock_held_time(trace),
+                    group.get_lock_wait_time(trace),
+                )
+                if not trace in traces.keys():
+                    traces[trace] = list[GroupTimes]()
+                traces.get(trace).append(group_time)
+
+        return traces
+
+
+def _make_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(description="spread filter helper")
+    parser.add_argument(
+        "-f", "--locks-file", type=str, required=True, help="Locks file"
+    )
+    parser.add_argument(
+        "--test",
+        default="",
+        help="Show results for this test, when test is selected no other filters are applied.",
+    )
+    parser.add_argument(
+        "--match",
+        action="append",
+        default=[],
+        help="Show traces that match this specific word",
+    )
+    parser.add_argument(
+        "--held-time", type=int, default=0, help="Include locks with longer held time"
+    )
+    parser.add_argument(
+        "--wait-time", type=int, default=0, help="Include locks with longer wait time"
+    )
+    parser.add_argument(
+        "--sort-held-time",
+        action="store_true",
+        help="Sort higher times first by held time",
+    )
+    parser.add_argument(
+        "--sort-wait-time",
+        action="store_true",
+        help="Sort higher times first by wait time",
+    )
+
+    return parser
+
+
+if __name__ == "__main__":
+    parser = _make_parser()
+    args = parser.parse_args()
+
+    if not os.path.isfile(args.locks_file):
+        print(
+            "state-lock-filter: locks file '{}' does not exist".format(args.locks_file)
+        )
+        sys.exit(1)
+
+    if args.sort_held_time and args.sort_wait_time:
+        print("state-lock-filter: define just 1 sorting (by held or wait times)")
+        sys.exit(1)
+
+    locks_reader = LocksFileReader(args.locks_file)
+    if args.test:
+        print(locks_reader.get_test(args.test))
+        sys.exit()
+
+    trace_manager = LockTraceManager(locks_reader.get_traces_times())
+
+    # Then keep traces with matches
+    if args.match:
+        trace_manager.match(args.match)
+
+    # First filter by time
+    if args.held_time > 0 or args.wait_time > 0:
+        trace_manager.filter(args.held_time, args.wait_time)
+
+    # And finally print the sorted results (if required)
+    trace_manager.print(args.sort_held_time, args.sort_wait_time)
```

Comment:
the overall code could use at least one unit test which should contain example state lock trace

Reply:
right, but perhaps if could be in a following pr

Reply:
I'll create a task to implement this

Reply:
perhaps a spread test in tools section?

Reply:
I think unit tests are generally better for this as we can use tools to look at coverage as we make changes locally.